astropy/coordinates/builtin_frames/galactocentric.py

```diff
@@ -149,59 +149,54 @@ def get_roll0(cls):
         return _ROLL0
 
 # ICRS to/from Galactocentric ----------------------->
-@frame_transform_graph.transform(FunctionTransform, ICRS, Galactocentric)
-def icrs_to_galactocentric(icrs_coord, galactocentric_frame):
-    if isinstance(icrs_coord.data, UnitSphericalRepresentation):
-        raise ConvertError("Transforming to a Galactocentric frame requires "
-                           "a 3D coordinate, e.g. (angle, angle, distance) or"
-                           " (x, y, z).")
+def get_matrix_vectors(galactocentric_frame):
+    # shorthand
+    gcf = galactocentric_frame
 
     # define rotation matrix to align x(ICRS) with the vector to the Galactic center
-    mat1 = rotation_matrix(-galactocentric_frame.galcen_dec, 'y')
-    mat2 = rotation_matrix(galactocentric_frame.galcen_ra, 'z')
+    mat1 = rotation_matrix(-gcf.galcen_dec, 'y')
+    mat2 = rotation_matrix(gcf.galcen_ra, 'z')
     # extra roll away from the Galactic x-z plane
-    mat0 = rotation_matrix(galactocentric_frame.get_roll0() - galactocentric_frame.roll, 'x')
+    mat0 = rotation_matrix(gcf.get_roll0() - gcf.roll, 'x')
 
     # construct transformation matrix and use it
     R = matrix_product(mat0, mat1, mat2)
-    intrep = icrs_coord.cartesian.transform(R)
 
     # Now need to translate by Sun-Galactic center distance around x' and
     # rotate about y' to account for tilt due to Sun's height above the plane
-    translation = CartesianRepresentation(galactocentric_frame.galcen_distance *
-                                          np.array([1., 0., 0.]))
-    z_d = (galactocentric_frame.z_sun / galactocentric_frame.galcen_distance).decompose()
-    rotation = rotation_matrix(-np.arcsin(z_d), 'y')
-    representation = (intrep - translation).transform(rotation)
+    translation = CartesianRepresentation(gcf.galcen_distance * [1., 0., 0.])
+    z_d = (gcf.z_sun / gcf.galcen_distance).decompose()
```

Comment:
Why the `decompose`? I'd think the `arcsin` would (effectively) do that for you.